### PR TITLE
created new job status endpoint

### PR DESCRIPTION
**Pull request by rachelhongca**
@sandlerben @anoshaminai HALP (I'm not sure if I did the func correctly)

### CODE REVIEW

**Comment by rachelhongca** [comment]
I also don't understand why it just failed the checks....

**Comment by yoninachmany** [comment]
maybe try merging master into ur branch if u haven't recently

**Comment by sandlerben** [comment]
r @yoninachmany 

After Yoni takes a look I can too

File: tasks/tasks.go [hunk 1]
```diff
@@ -33,6 +33,24 @@ const (
 	NOTFOUND
 )
 
+var DefaultTaskExecuter = NewTaskExectuer()
```
Comment (yoninachmany) [inline tasks/tasks.go]: `Executor` not `Executer` or `Exectuer`

File: tasks/tasks.go [hunk 2]
```diff
@@ -33,6 +33,24 @@ const (
...
+		str = "The task was successfully completed."
+	case FAILURE:
+		str = "The task failed."
+	case NOTFOUND:
+		str = "Error: not found"
+	}
+	return str
+}
+
 // NewTaskExectuer returns a TaskExecutor ready to execute.
```
Comment (yoninachmany) [inline tasks/tasks.go]: thought spelling was fixed in `task-queue` pr, but turns out not. `Executor` not `Exectuer`

File: tasks/tasks.go [hunk 3]
```diff
@@ -33,6 +33,24 @@ const (
...
 func NewTaskExectuer() TaskExecutor {
```
Comment (yoninachmany) [inline tasks/tasks.go]: `Executor` not `Exectuer`

File: tasks/tasks.go (hunk 1 above)
Comment (yoninachmany) [inline tasks/tasks.go]: @sandlerben looking in atom now, 
screen shot 2016-02-29 at 12 42 16 am

Can you add comment please?

File: web/routes.go [hunk 4]
```diff
@@ -63,3 +70,17 @@ func initiateTranscriptionJobHandler(w http.ResponseWriter, r *http.Request) {
 func healthHandler(w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte("healthy!"))
 }
+
+var statuses = [...]string{
```
Comment (yoninachmany) [inline web/routes.go]: what does `statuses` do?

File: web/routes.go [hunk 5]
```diff
@@ -63,3 +70,17 @@ func initiateTranscriptionJobHandler(w http.ResponseWriter, r *http.Request) {
...
+	"In progress",
+	"Done",
+	"Error",
+}
+
+func jobStatusHandler(w http.ResponseWriter, r *http.Request) {
```
Comment (yoninachmany) [inline web/routes.go:75]: @sandlerben thoughts on doc comments for these handler functions? Realized that I gave one on `initiateTranscriptionJobHandler', even though it's not exported - should we have doc comments for handler functions?

Comment (sandlerben) [inline web/routes.go:75]: I'm all for comments on unexported functions even if they aren't strictly required.

Comment (yoninachmany) [inline web/routes.go:75]: @rachelhongca please add a doc comment (for reference on Go comments, you can refer to: https://golang.org/doc/effective_go.html#commentary)

File: web/routes.go [hunk 6]
```diff
@@ -39,6 +40,12 @@ var routes = []route{
 		"/health",
 		healthHandler,
 	},
+	route{
+		"job_status",
```
Comment (yoninachmany) [inline web/routes.go:44]: @sandlerben how should `route.Name` look? for `/add_job`, i called the route `initiateTranscriptionJob` vs. rachel's way `/job_status` route called `job_status` - which do you prefer?

Comment (sandlerben) [inline web/routes.go:44]: I think either convention is good. Let's go with the convention we were already using for consistency.

Comment (yoninachmany) [inline web/routes.go:44]: kind of like rachel's way more, and it's only different in one place, so I will change. Thanks ben!

File: web/routes.go [hunk 7]
```diff
@@ -39,6 +40,12 @@ var routes = []route{
...
+		"GET",
+		"/job_status",
```
Comment (sandlerben) [inline web/routes.go]: I think we need to add something like `/{id}` to the end of the route to get this working.

**Comment by sandlerben** [comment]
@rachelhongca were you able to run this code to see if you can go to the job status endpoint? You can try loading `/job_status/kdaslfk` in your browser and ensuring that you get the appropriate ("task not found") response.

**Comment by yoninachmany** [comment]
LGTM. @sandlerben thoughts?

File: web/routes.go (hunk 4 above)
Comment (sandlerben) [inline web/routes.go]: Bump

Comment (rachelhongca) [inline web/routes.go]: oop sorry - I changed the way I did it so we shouldn't need this anymore.. I'll delete

File: web/routes.go [hunk 8]
```diff
@@ -63,3 +70,12 @@ func initiateTranscriptionJobHandler(w http.ResponseWriter, r *http.Request) {
 func healthHandler(w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte("healthy!"))
 }
+
+// jobStatusHandler takes a POST request and returns status message.
```
Comment (sandlerben) [inline web/routes.go]: It actually takes a get request. I don't think you need to specify the type of request it takes here. Instead write a comment like

``` go
// jobStatusHandler returns the status of a task with given id
```

File: tasks/tasks.go [hunk 9]
```diff
@@ -33,15 +33,34 @@ const (
...
+
+	switch s {
+	case INPROGRESS:
+		str = "The task is in progress."
+	case SUCCESS:
+		str = "The task was successfully completed."
+	case FAILURE:
+		str = "The task failed."
+	case NOTFOUND:
+		str = "Error: not found"
```
Comment (sandlerben) [inline tasks/tasks.go]: Change to `"Error: task not found."`

File: tasks/tasks.go [hunk 10]
```diff
@@ -33,15 +33,34 @@ const (
...
+var DefaultTaskExecuter = NewTaskExecuter()
+
+func (s Status) String() string {
+	var str string
+
+	switch s {
+	case INPROGRESS:
+		str = "The task is in progress."
+	case SUCCESS:
+		str = "The task was successfully completed."
```
Comment (sandlerben) [inline tasks/tasks.go:46]: Change to `"The task completed successfully"`

**Comment by sandlerben** [comment]
Left a few small comments, besides that looks good.

File: tasks/tasks.go (hunk 10 above)
Comment (sandlerben) [inline tasks/tasks.go:46]: @rachelhongca looks like you missed this comment

**Pull request merged**